persistence-api/src/main/java/io/stargate/db/DbActivator.java

```diff
@@ -0,0 +1,30 @@
+package io.stargate.db;
+
+import io.stargate.core.activator.BaseActivator;
+import io.stargate.db.datastore.DataStoreFactory;
+import java.util.Collections;
+import java.util.List;
+
+public class DbActivator extends BaseActivator {
+  private final ServicePointer<Persistence> persistence =
+      BaseActivator.ServicePointer.create(Persistence.class);
+
+  public DbActivator() {
+    super("DB services");
+  }
+
+  @Override
+  protected ServiceAndProperties createService() {
+    return new ServiceAndProperties(new DataStoreFactory(), DataStoreFactory.class);
```

Comment:
What is the rationale for registering `DataStoreFactory` as a service? The `DataStoreFactory` class appears to be self-sufficient. It is a concrete class without any configuration.... Bundles that use it should be able to create it whenever necessary without depending on other bundles :thinking:

Reply:
The intention is to have one common place where it is created - a singleton. Then, this one instance is injected wherever it is needed instead of using the static method and static init.

Reply:
Sorry, but I do not see any benefits of this. `DataStores` can be created from `Persistence` even now and without any static init AFAIK. This PR does not simplify the creation process, it seems - users still have to provide the same amount of information as before.

Reply:
> DataStores can be created from Persistence even now and without any static init AFAIK.
What do you mean? All places where DataStore is created are using `create()` method that is static.

Reply:
`create()` is indeed static, but in my mind it does not "initialize" anything, it's just a convenience method for creating a `DataStore`.

I did not see that other comment when I posted my comment avove :)

Having multiple `DataStore` implementations was not apparent from the PR :)

If we want to centralize `DataStore` creation, I'd propose defining an `interface` for that service rather than using a concrete class for that purpose.

Also since Stargate kind of assumes that only one `Persistence` implementation is active at any time. I believe it would be worth abstracting the dependency on `Persistence` behind the factory interface, so that callers could depend just on the facotory service. WDYT?

Reply:
Do you mean injecting the `Persistence` into this `DataStoreFactory` and remove the `Persistence` from all of it's create methods? That could be indeed a good improvement. However, from what I see, the `Persistence` is also used directly - no via the `DataStore` interface. Because of that, the `Persistence` cannot be hidden entirely from the callers. In the context of `DataStore` it can be hidden -  f that is enough for you, I may try to do it.

Reply:
Yes, that's what I mean.

I suppose not all `DataStore` users need to access `Persistence` directly... e.g. in GraphQL API it looks like in many places where `Persistence` is used, it is used only to create `DataStore`... I did not check all of the usages, but there's certainly a pattern there.

Reply:
I've moved the `Persistence` into `DataStore`. It looks better now. Please take a look.